include/openthread/platform/crypto.h

```diff
@@ -469,6 +469,36 @@ otError otPlatCryptoSha256Update(otCryptoContext *aContext, const void *aBuf, ui
  */
 otError otPlatCryptoSha256Finish(otCryptoContext *aContext, uint8_t *aHash, uint16_t aHashSize);
 
+/**
+ * Initialize cryptographically-secure pseudorandom number generator (CSPRNG).
+ *
+ * @retval OT_ERROR_NONE     Successfully initialized CSPRNG module.
+ * @retval OT_ERROR_FAILED   Failed to initialize CSPRNG module.
+ *
+ */
+otError otPlatCryptoRandomInit(void);
+
+/**
+ * Deinitialize cryptographically-secure pseudorandom number generator (CSPRNG).
+ *
+ * @retval OT_ERROR_NONE     Successfully deinitialized CSPRNG module.
+ * @retval OT_ERROR_FAILED   Failed to deinitialize CSPRNG module.
+ *
+ */
+otError otPlatCryptoRandomDeinit(void);
```

Comment:
This is fine as is, but I was thinking if it make sense to change all or some of these platform APIs to return `void` instead of `otError`.  For `Init()` and `DeInit()` we seems to `assert` on error within OT core, so making them `void` may be good (and can have platform implementation to assert on any failure). Even for `otPlatCryptoRandomGet()` in many cases we do assert on failure (in some case we ignore and in few cases we may retry the overall operation). Thoughts?

Reply:
I like this idea. It should be a straightforward decision for Inits and Deinits, for the rest of the operation, I guess we should look at OT core usage of those API. Would you like me to change return types in this PR or subsequent one?

Reply:
Thanks @LuDuda. I agree. Maybe in this PR we just change `Init()` and `Deinit()` to return `void` and leave the `RadomGet()` as is (we can cosnider changing that later maybe).